kafka-streams-partitioners/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/MultiLevelStreamsPartitioner.java

```diff
@@ -0,0 +1,58 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+import org.hypertrace.core.kafkastreams.framework.partitioner.MultiLevelPartitionerConfig.PartitionGroupConfig;
+
+/**
+ * Example config:
+ *
+ * <pre>
+ * groups.group1.members = tenant-1 # mandatory - for each configured group
+ * groups.group1.weight = 25
+ * groups.group2.members = tenant-2, tenant-3
+ * groups.group2.weight = 25
+ * default.group.weight = 50
+ * </pre>
+ */
+@Slf4j
+public class MultiLevelStreamsPartitioner<K, V> implements StreamPartitioner<K, V> {
```

Comment:
Likewise, I think we're combining two things here. If Kafka doesn't already provide it, having a Chained/MultiLevel partitioner that can keep delegating down makes sense, but I wouldn't put any partition logic in it.

So maybe we'd want to separate (or use, if it already exists) a MultiLevelPartitioner that takes a list of delegates and then also add a WeightedGroupPartitioner? Does that make sense? Or allow building it like:
```
ChainedPartitioner.of(new WeightedGroupPartitioner(...))
   .andThen(new SomeOtherPartitioner(...))
   .build();
```

Reply:
Discussed offline. Current  or near future requirement doesn't need a generic nested chaining.
Need more changes. Deferring now for this requirement.